_drafts/release-workflows.md

```diff
@@ -0,0 +1,388 @@
+| **merge** any branch to `4.0.0-beta`                           | Release a prerelease version (`4.0.0-beta`, then `4.0.0-beta.1`, then `4.0.0-beta.2`) on default channel (if supported, otherwise no channel) |
+| **push** to `5.0.0-beta`                                       | Release a prerelease version (`5.0.0-beta`, then `5.0.0-beta.1`, then `5.0.0-beta.2`) on default channel (if supported, otherwise no channel) |
+| **merge** any branch to `5.0.0-beta`                           | Release a prerelease version (`5.0.0-beta`, then `5.0.0-beta.1`, then `5.0.0-beta.2`) on default channel (if supported, otherwise no channel) |
+
+##### Pre-releases configuration
+
+```json
+"release": {
+  "branches": ["master", "4.0.0-beta", "5.0.0-alpha"]
+}
+```
+
+With specific branch names:
+
+```json
+"release": {
+  "branches": ["master", {
+    "branch": "dev",
+    "prerelease": "4.0.0-alpha"
+  }, {
+    "branch": "experimental",
+    "prerelease": "5.0.0-beta"
+  }]
+}
+```
+
+#### Configuration
+
+##### Format
+
+Each branch is either a `String` or an `Object` with the following properties:
+
+| Option       | Description                                                                                                                                                             |
+|--------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `branch`     | **Require** Git branch name                                                                                                                                             |
+| `range`      | Accepted version range for to release from this branch. If defined the branch will be considered a LTS branch.                                                          |
+| `channel`    | Channel on which to release. Ignored for release targets that don't support channels.                                                                                 |
+| `prerelease` | Version of the pre-release done from this branch. Must be formatted `<version>-<tag>`. The version is required, the `tag` is optional. Examples: `4.0.0-beta`, `4.0.0`. |
+
+**The order in which the branches are defines is meaningful, and it is used to determine the branch type and which release can be done from each branch.**
+
+##### Branches naming convention
+
+In a branch is defined as a `String` `semantic-release` will automatically determine the type of branch as follow:
+- If the branch name is a valid semver range (`1.x.x`, `1.0.x`, `^1.0.0`) the branch will be considered a LTS branch, with its range based on the branch name, and release on the channel with the same name formatted `1.x.x` (`^1.0.0` => `1.x.x`)
```

Comment:
I’m not sure about this. I feel we should only support the `1.x` format for LTS versions, that is the convention as far as I can tell. I can’t think of a reason where you would have branches `1.1.x` and `1.2.x`. And making `^` or `~` part of branch names ... I don’t know, I don’t think it’s a good idea to promote that

Reply:
`1.x === 1.x.x === ^1.0.0`. It's more simple for us to accept any valid form a semver range rather than limiting to a specific syntax.

We can use only `1.x` in the docs for more clarity.

Supporting `1.1.x` is useful if folks want to maintain a line of release that include a specific feature. It's a quite uncommon case, but thanks to the beauty of semver it's doesn't require us to do anything to support it.
There is no reason to add some code and complexity in order to prevent it.

Reply:
I feel rather strong about this. I understand it would be easy for use to to support any branch name that is a valid version range, but I don’t think it’s useful. I’d suggest we start with only supporting 1.x, 2.x etc and see how it goes. Adding support for other formats is always much easier than removing it

Reply:
Hehe, I feel quite strongly has well about it :) Because I prefer to use `semver.validRange` rather than getting into complex and difficult regex to limit to a certain form of a Semver range.
In that case, removing the support is way harder as we would have to re-code a part of the node semver module to detect which form is used.

But I agree we should mention only the `1.x` everywhere in the doc.

Reply:
the regex is `/^\d+\.x$/` :)

Reply:
Ok....But I really don't understand why we would go the extra miles to prevent users to do stuff, and make thinks less flexible. Especially when those things are consistent with the Semver spec that we embrace.

Semver supports multiple format of ranges. That might be confusing or useless but that's how the spec is. If we don't like we should open an issue with the Semver to propose the change. I don't see why semantic-release would have to do some extra work to exclude some part of the spec.

Reply:
It’s much easier to explain that semantic-release supports branch names in the formant `<first version number>.x` then explaining it supports all branch names that are also valid semver strings. Because

1. People have to understand what valid semver range strings are
2. I don’t see a use case where I would event want to support a branch name like `~2.8.0`

It’s not only about the users that setup semantic-release, but other contributors who come to a project, they also need to understand what these branches are. And the more projects follow the `1.x`, `2.x` convention, the more recognizable they will be, the easier it will be for folks to understand what they are

Reply:
btw `^1.0.0` and `~1.0.0` are not valid branch names as Felix pointed out at https://gist.github.com/pvdlg/9fa33dbc6792edc2cf3d0765c5d8b8b3#gistcomment-2302042